lib/Dialect/Handshake/MemoryInterfaces.cpp

```diff
@@ -387,19 +388,37 @@ void LSQGenerationInfo::fromPorts(FuncMemoryPorts &ports) {
     loadsPerGroup.push_back(groupPorts.getNumPorts<LoadPort>());
     storesPerGroup.push_back(groupPorts.getNumPorts<StorePort>());
 
-    // Compute the ffset of first load/store in the group and indices of each
-    // load/store port
+    // Track the numebr of stores within a group
+    unsigned numStores = 0;
+
+    // Compute the ffset of first load/store in the group and indices of
+    // each load/store port
     std::optional<unsigned> firstLoadOffset, firstStoreOffset;
     SmallVector<unsigned> groupLoadPorts, groupStorePorts;
+    unsigned numEntries = groupPorts.getNumPorts<StorePort>()
+                              ? groupPorts.getNumPorts<StorePort>()
+                              : 1;
```

Comment:
`numStoreEntries` since you only care about the number of stores here? Maybe use max(groupPorts.getNumPorts<StorePort>, 1) instead.